// @storybook/components/dist/navigation/RoutedLink.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LEFT_BUTTON = 0;
// Cmd/Ctrl/Shift/Alt + Click should trigger default browser behaviour. Same applies to non-left clicks
var isPlainLeftClick = function isPlainLeftClick(e) {
  return e.button === LEFT_BUTTON && !e.altKey && !e.ctrlKey && !e.metaKey && !e.shiftKey;
};

var RoutedLink = function (_React$Component) {
  (0, _inherits3.default)(RoutedLink, _React$Component);

  function RoutedLink() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, RoutedLink);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = RoutedLink.__proto__ || (0, _getPrototypeOf2.default)(RoutedLink)).call.apply(_ref, [this].concat(args))), _this), _this.onClick = function (e) {
      var onClick = _this.props.onClick;


      if (isPlainLeftClick(e)) {
        e.preventDefault();
        onClick(e);
      }
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(RoutedLink, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          href = _props.href,
          children = _props.children,
          onClick = _props.onClick,
          className = _props.className,
          style = _props.style;

      var props = onClick ? { href: href, className: className, style: style, onClick: this.onClick } : { href: href, className: className, style: style };
      return _react2.default.createElement(
        'a',
        props,
        children
      );
    }
  }]);
  return RoutedLink;
}(_react2.default.Component);

exports.default = RoutedLink;


RoutedLink.defaultProps = {
  onClick: null,
  href: '#',
  children: null,
  className: undefined,
  style: undefined
};

RoutedLink.propTypes = {
  onClick: _propTypes2.default.func,
  href: _propTypes2.default.string,
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  // eslint-disable-next-line react/forbid-prop-types
  style: _propTypes2.default.object
};
