// @storybook/components/dist/form/field.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactEmotion = require('react-emotion');

var _reactEmotion2 = _interopRequireDefault(_reactEmotion);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Wrapper = (0, _reactEmotion2.default)('label')(function (_ref) {
  var theme = _ref.theme;
  return {
    width: '100%',
    boxSizing: 'border-box',
    display: 'flex',
    margin: '8px 0',
    borderBottom: theme.mainBorder,
    padding: '0 8px 8px 8px',

    '&:last-child': {
      borderBottom: '0 none',
      paddingBottom: 0
    }
  };
});

var Label = (0, _reactEmotion2.default)('span')({
  minWidth: 100,
  minHeight: 32,
  marginRight: 16,
  display: 'flex',
  justifyContent: 'flex-start',
  alignItems: 'center',
  lineHeight: '16px'
});

var Field = function Field(_ref2) {
  var label = _ref2.label,
      children = _ref2.children;
  return _react2.default.createElement(
    Wrapper,
    null,
    label ? _react2.default.createElement(
      Label,
      null,
      _react2.default.createElement(
        'span',
        null,
        label
      )
    ) : null,
    children
  );
};
Field.propTypes = {
  label: _propTypes2.default.node,
  children: _propTypes2.default.node.isRequired
};
Field.defaultProps = {
  label: undefined
};

exports.default = Field;
