// @storybook/core/dist/server/config/webpack.config.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _toConsumableArray2 = require('babel-runtime/helpers/toConsumableArray');

var _toConsumableArray3 = _interopRequireDefault(_toConsumableArray2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _path = require('path');

var _path2 = _interopRequireDefault(_path);

var _webpack = require('webpack');

var _webpack2 = _interopRequireDefault(_webpack);

var _universalDotenv = require('universal-dotenv');

var _dotenvWebpack = require('dotenv-webpack');

var _dotenvWebpack2 = _interopRequireDefault(_dotenvWebpack);

var _InterpolateHtmlPlugin = require('react-dev-utils/InterpolateHtmlPlugin');

var _InterpolateHtmlPlugin2 = _interopRequireDefault(_InterpolateHtmlPlugin);

var _WatchMissingNodeModulesPlugin = require('react-dev-utils/WatchMissingNodeModulesPlugin');

var _WatchMissingNodeModulesPlugin2 = _interopRequireDefault(_WatchMissingNodeModulesPlugin);

var _caseSensitivePathsWebpackPlugin = require('case-sensitive-paths-webpack-plugin');

var _caseSensitivePathsWebpackPlugin2 = _interopRequireDefault(_caseSensitivePathsWebpackPlugin);

var _htmlWebpackPlugin = require('html-webpack-plugin');

var _htmlWebpackPlugin2 = _interopRequireDefault(_htmlWebpackPlugin);

var _utils = require('../utils');

var _utils2 = require('./utils');

var _package = require('../../../package.json');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function (_ref) {
  var configDir = _ref.configDir,
      quiet = _ref.quiet,
      babelOptions = _ref.babelOptions;

  var entries = (0, _utils2.getEntries)(configDir, true);
  return {
    mode: 'development',
    devtool: 'cheap-module-source-map',
    entry: (0, _extends3.default)({}, entries, {
      preview: [].concat((0, _toConsumableArray3.default)(entries.preview), [require.resolve('webpack-hot-middleware/client') + '?reload=true'])
    }),
    output: {
      path: _path2.default.join(__dirname, 'dist'),
      filename: 'static/[name].bundle.js',
      // Here we set the publicPath to ''.
      // This allows us to deploy storybook into subpaths like GitHub pages.
      // This works with css and image loaders too.
      // This is working for storybook since, we don't use pushState urls and
      // relative URLs works always.
      publicPath: ''
    },
    plugins: [new _htmlWebpackPlugin2.default({
      filename: 'index.html',
      chunks: ['manager'],
      chunksSortMode: 'none',
      data: {
        managerHead: (0, _utils.getManagerHeadHtml)(configDir),
        version: _package.version
      },
      template: require.resolve('../index.html.ejs')
    }), new _htmlWebpackPlugin2.default({
      filename: 'iframe.html',
      excludeChunks: ['manager'],
      chunksSortMode: 'none',
      data: {
        previewHead: (0, _utils.getPreviewHeadHtml)(configDir)
      },
      template: require.resolve('../iframe.html.ejs')
    }), new _InterpolateHtmlPlugin2.default(process.env), new _webpack2.default.DefinePlugin((0, _utils2.loadEnv)()), new _webpack2.default.HotModuleReplacementPlugin(), new _caseSensitivePathsWebpackPlugin2.default(), new _WatchMissingNodeModulesPlugin2.default(_utils2.nodeModulesPaths), quiet ? null : new _webpack2.default.ProgressPlugin(), new _webpack2.default.DefinePlugin((0, _universalDotenv.getEnvironment)().webpack), new _dotenvWebpack2.default({ silent: true })].filter(Boolean),
    module: {
      rules: [{
        test: /\.js$/,
        use: [{
          loader: require.resolve('babel-loader'),
          options: babelOptions
        }],
        include: _utils2.includePaths,
        exclude: _utils2.excludePaths
      }, {
        test: /\.md$/,
        use: [{
          loader: require.resolve('raw-loader')
        }]
      }]
    },
    resolve: {
      // Since we ship with json-loader always, it's better to move extensions to here
      // from the default config.
      extensions: ['.js', '.json'],
      // Add support to NODE_PATH. With this we could avoid relative path imports.
      // Based on this CRA feature: https://github.com/facebookincubator/create-react-app/issues/253
      modules: ['node_modules'].concat(_utils2.nodePaths)
    },
    performance: {
      hints: false
    }
  };
};
