// @storybook/components/dist/panel_actionbar/panel_actionbar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionBar = exports.ActionItem = exports.ActionButton = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactEmotion = require('react-emotion');

var _reactEmotion2 = _interopRequireDefault(_reactEmotion);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Container = (0, _reactEmotion2.default)('ul')(function (_ref) {
  var theme = _ref.theme;
  return {
    position: 'absolute',
    bottom: 0,
    right: 0,
    maxWidth: '100%',
    display: 'flex',
    listStyle: 'none',
    margin: 0,
    padding: 0,

    borderTop: theme.mainBorder,
    borderLeft: theme.mainBorder,
    background: theme.barFill,
    borderRadius: theme.mainBorderRadius + 'px 0 0 0'
  };
});

var ActionButton = exports.ActionButton = (0, _reactEmotion2.default)('button')(function (_ref2) {
  var theme = _ref2.theme;
  return {
    border: '0 none',
    display: 'block',
    background: 'none',
    padding: '0 10px',

    textTransform: 'inherit',
    letterSpacing: 'inherit',
    fontSize: 'inherit',
    color: 'inherit',

    borderTop: '2px solid transparent',
    borderBottom: '2px solid transparent',

    '&:focus': {
      borderBottom: '2px solid ' + theme.highlightColor,
      outline: '0 none'
    }
  };
});
ActionButton.displayName = 'ActionButton';

var ActionItem = exports.ActionItem = (0, _reactEmotion2.default)('li')(function (_ref3) {
  var first = _ref3.first,
      theme = _ref3.theme;
  return {
    display: 'flex',
    position: 'relative',
    listStyle: 'none',
    padding: 0,
    height: 26,
    fontSize: 11,
    letterSpacing: 1,
    textTransform: 'uppercase',

    borderLeft: first ? '0 none' : '1px solid ' + theme.mainBorderColor
  };
});
ActionItem.displayName = 'ActionItem';

var ActionBar = exports.ActionBar = function ActionBar(_ref4) {
  var children = _ref4.children;
  return _react2.default.createElement(
    Container,
    null,
    _react.Children.toArray(children).map(function (item, index) {
      return (
        // eslint-disable-next-line react/no-array-index-key
        _react2.default.createElement(
          ActionItem,
          { key: index, first: index === 0 },
          item
        )
      );
    })
  );
};

ActionBar.propTypes = {
  children: _propTypes2.default.node.isRequired
};
