// @storybook/components/dist/addon_panel/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _slicedToArray2 = require('babel-runtime/helpers/slicedToArray');

var _slicedToArray3 = _interopRequireDefault(_slicedToArray2);

var _entries = require('babel-runtime/core-js/object/entries');

var _entries2 = _interopRequireDefault(_entries);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _tabs = require('../tabs/tabs');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AddonPanel = function AddonPanel(_ref) {
  var panels = _ref.panels,
      onPanelSelect = _ref.onPanelSelect,
      selectedPanel = _ref.selectedPanel,
      rest = (0, _objectWithoutProperties3.default)(_ref, ['panels', 'onPanelSelect', 'selectedPanel']);
  return _react2.default.createElement(
    _tabs.Tabs,
    (0, _extends3.default)({}, rest, { absolute: true, bordered: true, selected: selectedPanel, onSelect: onPanelSelect }),
    (0, _entries2.default)(panels).map(function (_ref2) {
      var _ref3 = (0, _slicedToArray3.default)(_ref2, 2),
          k = _ref3[0],
          v = _ref3[1];

      return _react2.default.createElement(
        'div',
        { key: k, id: k, title: v.title },
        v.render
      );
    })
  );
};
AddonPanel.propTypes = {
  selectedPanel: _propTypes2.default.string,
  onPanelSelect: _propTypes2.default.func.isRequired,
  panels: _propTypes2.default.objectOf(_propTypes2.default.shape({
    title: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func]),
    render: _propTypes2.default.func
  })).isRequired
};
AddonPanel.defaultProps = {
  selectedPanel: null
};

exports.default = AddonPanel;
