// @storybook/components/dist/navigation/MenuLink.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactEmotion = require('react-emotion');

var _reactEmotion2 = _interopRequireDefault(_reactEmotion);

var _RoutedLink = require('./RoutedLink');

var _RoutedLink2 = _interopRequireDefault(_RoutedLink);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var MenuLink = (0, _reactEmotion2.default)(_RoutedLink2.default, { rootEl: 'a' })(function (_ref) {
  var theme = _ref.theme;
  return {
    display: 'block',
    color: theme.dimmedTextColor,
    textDecoration: 'none',
    fontSize: '13px',
    lineHeight: '16px',
    padding: '1px 5px 4px',
    marginLeft: '5px',
    position: 'relative',
    zIndex: 1
  };
}, function (_ref2) {
  var active = _ref2.active;
  return active ? {
    color: 'inherit',
    fontWeight: 'bold',
    backgroundColor: 'rgba(0,0,0,0.07)',
    zIndex: 0
  } : {};
});
MenuLink.propTypes = {
  active: _propTypes2.default.bool
};

exports.default = MenuLink;
