// @storybook/components/dist/header/header.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactEmotion = require('react-emotion');

var _reactEmotion2 = _interopRequireDefault(_reactEmotion);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Wrapper = (0, _reactEmotion2.default)('div')(function (_ref) {
  var theme = _ref.theme;
  return (0, _extends3.default)({
    margin: '0 0 10px',
    display: 'flex'
  }, theme.brand);
});

var HeadingLink = (0, _reactEmotion2.default)('a')({
  textDecoration: 'none',
  flexGrow: 1,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  border: '1px solid rgba(0, 0, 0, 0.1)',
  borderRadius: 2,
  textTransform: 'uppercase',
  letterSpacing: '1.5px',
  fontSize: '12px',
  fontWeight: 'bolder',
  color: 'currentColor',
  textAlign: 'center',
  cursor: 'pointer',
  padding: '5px',
  margin: 0,
  overflow: 'hidden'
});

var ShortHelpButton = (0, _reactEmotion2.default)('button')({
  textTransform: 'uppercase',
  fontSize: 12,
  fontWeight: 'bolder',
  color: 'currentColor',
  border: '1px solid rgba(0, 0, 0, 0.1)',
  textAlign: 'center',
  borderRadius: 2,
  cursor: 'pointer',
  display: 'inlineBlock',
  padding: 0,
  margin: '0 0 0 5px',
  backgroundColor: 'inherit',
  outline: 0,
  width: 30,
  flexShrink: 0
});

var Header = function Header(_ref2) {
  var openShortcutsHelp = _ref2.openShortcutsHelp,
      name = _ref2.name,
      url = _ref2.url,
      enableShortcutsHelp = _ref2.enableShortcutsHelp,
      isMobileDevice = _ref2.isMobileDevice;
  return _react2.default.createElement(
    Wrapper,
    { isMobileDevice: isMobileDevice },
    _react2.default.createElement(
      HeadingLink,
      { href: url, target: '_blank', rel: 'noopener noreferrer' },
      name
    ),
    enableShortcutsHelp && _react2.default.createElement(
      ShortHelpButton,
      { onClick: openShortcutsHelp },
      '\u2318'
    )
  );
};

Header.defaultProps = {
  openShortcutsHelp: null,
  enableShortcutsHelp: true,
  name: '',
  url: '',
  isMobileDevice: false
};

Header.propTypes = {
  openShortcutsHelp: _propTypes2.default.func,
  name: _propTypes2.default.string,
  url: _propTypes2.default.string,
  isMobileDevice: _propTypes2.default.bool,
  enableShortcutsHelp: _propTypes2.default.bool
};

exports.default = Header;
