// @storybook/components/dist/tabs/tabs.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TabsState = exports.Tabs = exports.panelProps = exports.TabWrapper = exports.Tab = exports.TabBar = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _renderFragment = require('render-fragment');

var _renderFragment2 = _interopRequireDefault(_renderFragment);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactEmotion = require('react-emotion');

var _reactEmotion2 = _interopRequireDefault(_reactEmotion);

var _placeholder = require('../placeholder/placeholder');

var _placeholder2 = _interopRequireDefault(_placeholder);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Wrapper = (0, _reactEmotion2.default)('div')(function (_ref) {
  var theme = _ref.theme,
      bordered = _ref.bordered;
  return bordered ? {
    overflow: 'hidden',
    background: theme.mainFill,
    borderRadius: theme.mainBorderRadius,
    border: theme.mainBorder
  } : {};
}, function (_ref2) {
  var absolute = _ref2.absolute;
  return absolute ? {
    width: '100%',
    height: '100%',
    boxSizing: 'border-box',
    display: 'flex',
    flexDirection: 'column'
  } : {
    display: 'block'
  };
});

var TabBar = exports.TabBar = (0, _reactEmotion2.default)('div')(function (_ref3) {
  var theme = _ref3.theme;
  return {
    background: theme.barFill,
    borderBottom: theme.mainBorder,
    overflow: 'auto',
    padding: '0 8px',
    whiteSpace: 'nowrap',
    height: 40
  };
});

var TabButton = (0, _reactEmotion2.default)('button')({
  whiteSpace: 'normal',
  display: 'inline-flex',
  overflow: 'hidden',
  verticalAlign: 'top',
  justifyContent: 'center',
  alignItems: 'center',
  textAlign: 'center',

  '&:empty': {
    display: 'none'
  }
}, function (_ref4) {
  var theme = _ref4.theme;
  return {
    fontSize: 11,
    letterSpacing: '1px',
    padding: '0 8px',
    textTransform: 'uppercase',
    transition: 'color 0.2s linear, border-bottom-color 0.2s linear',
    height: 40,
    lineHeight: '12px',
    boxSizing: 'border-box',
    cursor: 'pointer',
    background: 'transparent',
    border: '0 solid transparent',
    borderTop: '3px solid transparent',
    borderBottom: '3px solid transparent',

    '&:focus': {
      outline: '0 none',
      borderBottomColor: theme.highlightColor
    }
  };
}, function (_ref5) {
  var active = _ref5.active,
      theme = _ref5.theme;
  return active ? {
    color: theme.mainTextColor,
    borderBottomColor: theme.barSelectedColor
  } : {
    color: theme.dimmedTextColor,
    borderBottomColor: 'transparent'
  };
});

var Content = (0, _reactEmotion2.default)('div')({
  display: 'block',
  position: 'relative'
}, function (_ref6) {
  var absolute = _ref6.absolute;
  return absolute ? {
    position: 'relative',
    overflow: 'auto',
    flex: 1,
    width: '100%'
  } : {};
});

var Tab = exports.Tab = function Tab(_ref7) {
  var active = _ref7.active,
      name = _ref7.name,
      title = _ref7.title,
      onSelect = _ref7.onSelect;

  var onClick = function onClick(e) {
    e.preventDefault();
    onSelect(name);
  };

  return _react2.default.createElement(
    TabButton,
    { type: 'button', key: name, active: active, onClick: onClick, role: 'tab' },
    typeof title === 'function' ? title() : title
  );
};

Tab.propTypes = {
  active: _propTypes2.default.bool.isRequired,
  name: _propTypes2.default.string.isRequired,
  title: _propTypes2.default.oneOfType([_propTypes2.default.func, _propTypes2.default.node]).isRequired,
  onSelect: _propTypes2.default.func.isRequired
};

var VisuallyHidden = (0, _reactEmotion2.default)('div')(function (_ref8) {
  var active = _ref8.active;
  return active ? { display: 'block' } : { display: 'none' };
});
var TabWrapper = exports.TabWrapper = function TabWrapper(_ref9) {
  var active = _ref9.active,
      render = _ref9.render,
      children = _ref9.children;
  return _react2.default.createElement(
    VisuallyHidden,
    { active: active },
    render ? render() : children
  );
};
TabWrapper.propTypes = {
  active: _propTypes2.default.bool.isRequired,
  render: _propTypes2.default.func,
  children: _propTypes2.default.node
};
TabWrapper.defaultProps = {
  render: undefined,
  children: undefined
};

var panelProps = exports.panelProps = {
  active: _propTypes2.default.bool
};

var Tabs = exports.Tabs = function Tabs(_ref10) {
  var children = _ref10.children,
      selected = _ref10.selected,
      onSelect = _ref10.onSelect,
      absolute = _ref10.absolute,
      bordered = _ref10.bordered;

  var list = _react2.default.Children.toArray(children).map(function (_ref11, index) {
    var _ref11$props = _ref11.props,
        title = _ref11$props.title,
        id = _ref11$props.id,
        childrenOfChild = _ref11$props.children;

    var content = Array.isArray(childrenOfChild) ? childrenOfChild[0] : childrenOfChild;
    return {
      active: selected ? id === selected : index === 0,
      title: title,
      id: id,
      render: typeof content === 'function' ? content : // eslint-disable-next-line react/prop-types
      function (_ref12) {
        var active = _ref12.active;
        return _react2.default.createElement(
          VisuallyHidden,
          { active: active, role: 'tabpanel' },
          content
        );
      }
    };
  });

  return list.length ? _react2.default.createElement(
    Wrapper,
    { absolute: absolute, bordered: bordered },
    _react2.default.createElement(
      TabBar,
      { role: 'tablist' },
      list.map(function (_ref13) {
        var title = _ref13.title,
            id = _ref13.id,
            active = _ref13.active;
        return _react2.default.createElement(Tab, { key: id + '-bar', active: active, name: id, title: title, onSelect: onSelect });
      })
    ),
    _react2.default.createElement(
      Content,
      { absolute: absolute },
      list.map(function (_ref14) {
        var id = _ref14.id,
            active = _ref14.active,
            render = _ref14.render;
        return _react2.default.createElement(
          _renderFragment2.default,
          { key: id + '-panel' },
          render({
            active: active,
            selected: selected
          })
        );
      })
    )
  ) : _react2.default.createElement(
    _placeholder2.default,
    null,
    'no panels available'
  );
};
Tabs.propTypes = {
  children: _propTypes2.default.arrayOf(_propTypes2.default.node),
  selected: _propTypes2.default.string,
  onSelect: _propTypes2.default.func.isRequired,
  absolute: _propTypes2.default.bool.isRequired,
  bordered: _propTypes2.default.bool.isRequired
};
Tabs.defaultProps = {
  children: [],
  selected: null
};

var TabsState = exports.TabsState = function (_Component) {
  (0, _inherits3.default)(TabsState, _Component);

  function TabsState(props) {
    (0, _classCallCheck3.default)(this, TabsState);

    var _this = (0, _possibleConstructorReturn3.default)(this, (TabsState.__proto__ || (0, _getPrototypeOf2.default)(TabsState)).call(this, props));

    _this.state = {
      selected: props.initial
    };
    return _this;
  }

  (0, _createClass3.default)(TabsState, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          bordered = _props.bordered,
          absolute = _props.absolute,
          children = _props.children;
      var selected = this.state.selected;

      return _react2.default.createElement(
        Tabs,
        {
          bordered: bordered || true,
          absolute: absolute || false,
          selected: selected,
          onSelect: function onSelect(id) {
            return _this2.setState({ selected: id });
          }
        },
        children
      );
    }
  }]);
  return TabsState;
}(_react.Component);

TabsState.propTypes = {
  children: _propTypes2.default.arrayOf(_propTypes2.default.node),
  initial: _propTypes2.default.string,
  absolute: _propTypes2.default.bool,
  bordered: _propTypes2.default.bool
};
TabsState.defaultProps = {
  children: [],
  initial: null,
  absolute: false,
  bordered: false
};
