// @storybook/components/dist/layout/dimensions.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactEmotion = require('react-emotion');

var _reactEmotion2 = _interopRequireDefault(_reactEmotion);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DISPLAY_TIMEOUT = 1000;

var Container = (0, _reactEmotion2.default)('div')({
  position: 'absolute',
  padding: '5px 8px',
  bottom: 0,
  right: 0,
  backgroundColor: 'rgba(255, 255, 255, 0.5)',
  borderRadius: '5px 0 0 0'
});

var Section = (0, _reactEmotion2.default)('span')({ fontSize: 12 });

var Delimiter = (0, _reactEmotion2.default)('span')({
  margin: '0px 5px',
  fontSize: 12
});

var Dimensions = function (_Component) {
  (0, _inherits3.default)(Dimensions, _Component);

  function Dimensions(props) {
    (0, _classCallCheck3.default)(this, Dimensions);

    var _this = (0, _possibleConstructorReturn3.default)(this, (Dimensions.__proto__ || (0, _getPrototypeOf2.default)(Dimensions)).call(this, props));

    _this.state = (0, _extends3.default)({}, props, {
      isVisible: false
    });

    _this.hideTimeout = null;
    return _this;
  }

  (0, _createClass3.default)(Dimensions, [{
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      var _this2 = this;

      var isVisible = this.state.isVisible;


      if (isVisible) {
        clearTimeout(this.hideTimeout);
        this.hideTimeout = setTimeout(function () {
          _this2.setState({ isVisible: false });
        }, DISPLAY_TIMEOUT);
      }
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      clearTimeout(this.hideTimeout);
    }
  }, {
    key: 'render',
    value: function render() {
      var isVisible = this.state.isVisible;
      var _props = this.props,
          width = _props.width,
          height = _props.height;


      return isVisible ? _react2.default.createElement(
        Container,
        null,
        _react2.default.createElement(
          Section,
          null,
          width + 'px'
        ),
        _react2.default.createElement(
          Delimiter,
          null,
          'x'
        ),
        _react2.default.createElement(
          Section,
          null,
          height + 'px'
        )
      ) : null;
    }
  }]);
  return Dimensions;
}(_react.Component);

Dimensions.getDerivedStateFromProps = function (nextProps, prevState) {
  if (nextProps.width !== prevState.width || nextProps.height !== prevState.height) {
    return (0, _extends3.default)({}, nextProps, { isVisible: true });
  }
  return null;
};

Dimensions.propTypes = {
  width: _propTypes2.default.number.isRequired,
  height: _propTypes2.default.number.isRequired
};

exports.default = Dimensions;
