// @storybook/components/dist/layout/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Root = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactEmotion = require('react-emotion');

var _reactEmotion2 = _interopRequireDefault(_reactEmotion);

var _emotionTheming = require('emotion-theming');

var _theme = require('../theme');

var _mobile = require('./mobile');

var _mobile2 = _interopRequireDefault(_mobile);

var _desktop = require('./desktop');

var _desktop2 = _interopRequireDefault(_desktop);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Root = exports.Root = (0, _reactEmotion2.default)('div')(function (_ref) {
  var theme = _ref.theme;
  return {
    background: theme.mainBackground,
    fontFamily: theme.mainTextFace,
    color: theme.mainTextColor,
    fontSize: theme.mainTextSize
  };
}, function (_ref2) {
  var isMobileDevice = _ref2.isMobileDevice;
  return isMobileDevice ? {
    display: 'block',
    position: 'fixed',
    top: 0,
    left: 0,
    height: '100vh',
    width: '100vw',
    overflow: 'auto'
  } : {
    position: 'fixed',
    top: 0,
    left: 0,
    height: '100vh',
    width: '100vw',
    overflow: 'hidden'
  };
});

var StoriesPanelInner = (0, _reactEmotion2.default)('div')({
  flexGrow: 1,
  position: 'relative',
  height: '100%',
  width: '100%',
  overflow: 'auto'
});

var Layout = function Layout(props) {
  var isMobileDevice = props.isMobileDevice,
      theme = props.theme,
      AddonPanel = props.addonPanel,
      StoriesPanel = props.storiesPanel,
      Preview = props.preview,
      ShortcutsHelp = props.shortcutsHelp,
      SearchBox = props.searchBox;


  return _react2.default.createElement(
    _emotionTheming.ThemeProvider,
    { theme: theme || _theme.normal },
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(
        Root,
        { isMobileDevice: isMobileDevice },
        isMobileDevice ? _react2.default.createElement(
          _mobile2.default,
          null,
          _react2.default.createElement(
            StoriesPanelInner,
            null,
            _react2.default.createElement(StoriesPanel, null)
          ),
          _react2.default.createElement(
            StoriesPanelInner,
            null,
            _react2.default.createElement(Preview, null)
          ),
          _react2.default.createElement(AddonPanel, null)
        ) : _react2.default.createElement(_desktop2.default, props)
      ),
      _react2.default.createElement(ShortcutsHelp, null),
      _react2.default.createElement(SearchBox, null)
    )
  );
};

Layout.propTypes = {
  theme: _propTypes2.default.shape({}),
  showStoriesPanel: _propTypes2.default.bool.isRequired,
  showAddonPanel: _propTypes2.default.bool.isRequired,
  goFullScreen: _propTypes2.default.bool.isRequired,
  storiesPanel: _propTypes2.default.func.isRequired,
  preview: _propTypes2.default.func.isRequired,
  addonPanel: _propTypes2.default.func.isRequired,
  shortcutsHelp: _propTypes2.default.func.isRequired,
  searchBox: _propTypes2.default.func.isRequired,
  addonPanelInRight: _propTypes2.default.bool.isRequired,
  isMobileDevice: _propTypes2.default.bool.isRequired
};

Layout.defaultProps = {
  theme: null
};

exports.default = Layout;
