// @storybook/components/dist/layout/mobile.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactEmotion = require('react-emotion');

var _reactEmotion2 = _interopRequireDefault(_reactEmotion);

var _tabs = require('../tabs/tabs');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var MobilePanel = (0, _reactEmotion2.default)('div')(function (_ref) {
  var selected = _ref.selected;
  return selected ? {
    display: 'block',
    position: 'fixed',
    top: 0,
    left: 0,
    height: 'calc(100vh - 40px)',
    width: '100vw',
    overflow: 'auto',
    WebkitOverflowScrolling: 'touch'
  } : {
    display: 'none'
  };
});
var TabsWrapper = (0, _reactEmotion2.default)('div')({
  position: 'fixed',
  bottom: 0,
  left: 0,
  width: '100vw',
  zIndex: 9999
});

var MobileTabs = ['Navigator', 'Preview', 'Addons'];

var Layout = function (_Component) {
  (0, _inherits3.default)(Layout, _Component);

  function Layout() {
    var _ref2;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, Layout);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref2 = Layout.__proto__ || (0, _getPrototypeOf2.default)(Layout)).call.apply(_ref2, [this].concat(args))), _this), _this.state = { index: 1 }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(Layout, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var index = this.state.index;
      var children = this.props.children;


      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          TabsWrapper,
          null,
          _react2.default.createElement(
            _tabs.TabBar,
            null,
            children.map(function (item, i) {
              return _react2.default.createElement(_tabs.Tab, {
                key: MobileTabs[i],
                name: MobileTabs[i],
                active: i === index,
                title: MobileTabs[i] || '' + i,
                onSelect: function onSelect() {
                  return _this2.setState({ index: i });
                }
              });
            })
          )
        ),
        _react.Children.toArray(children).map(function (item, i) {
          return _react2.default.createElement(
            MobilePanel,
            { key: MobileTabs[i], selected: i === index },
            item
          );
        })
      );
    }
  }]);
  return Layout;
}(_react.Component);

exports.default = Layout;

Layout.propTypes = {
  children: _propTypes2.default.node.isRequired
};
